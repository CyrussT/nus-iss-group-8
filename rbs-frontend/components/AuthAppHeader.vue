<script setup lang="ts">
const auth = useAuthStore()
const router = useRouter()
const colorMode = useColorMode()
const showManageDropdown = ref(false)

const toggleColorMode = () => {
  colorMode.preference = colorMode.value === "dark" ? "light" : "dark";
};

const userRole = ref<string | null>(null)

watch(
  () => auth.user.value?.role,
  (newRole) => {
    userRole.value = newRole || null
  },
  { immediate: true }
)

const isAdmin = computed(() => userRole.value === 'ADMINISTRATOR')
const isStudent = computed(() => userRole.value === 'STUDENT')

const handleLogout = () => {
  auth.logout()
  router.push('/login')
}

interface NavigationMenuItem {
  label: string
  to?: string
  active?: boolean
  target?: '_blank' | '_self' | '_parent' | '_top'
}

const items = computed<NavigationMenuItem[]>(() => {
  const baseItems: NavigationMenuItem[] = []

  if (isAdmin.value) {
    baseItems.push(
      { label: 'Facility', to: '/facility' },
      { label: 'Manage Students', to: '/student-list' },
      { label: 'Booking Requests', to: '/booking-review-list' }
    )
  }

  if (isStudent.value) {
    baseItems.push({ label: 'Booking', to: '/booking' })
  }

  return baseItems
})
</script>

<template>
  <header class="h-16 border-b px-4 flex items-center justify-between">
    <h1 class="font-bold text-xl">
      <NuxtLink to="/">Resource Booking System</NuxtLink>
    </h1>

    <!-- Navigation -->
    <UHorizontalNavigation :links="items" class="justify-center" />

    <!-- Right controls -->
    <div class="flex items-center gap-4">
      <UButton variant="ghost" @click="toggleColorMode">
        <UIcon
          v-if="colorMode.preference === 'light'"
          name="i-heroicons:moon"
          class="w-5 h-5"
        />
        <UIcon
          v-else
          name="i-heroicons:sun"
          class="w-5 h-5"
        />
      </UButton>

      <UButton :ui="{ rounded: 'rounded-full' }" size="lg" @click="handleLogout">
        Logout
      </UButton>
    </div>
  </header>
</template>
