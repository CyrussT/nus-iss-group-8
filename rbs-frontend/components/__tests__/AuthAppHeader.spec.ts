import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import AuthAppHeader from '../AuthAppHeader.vue';

describe('AuthAppHeader', () => {
  it('renders correctly', () => {
    const wrapper = mount(AuthAppHeader, {
      global: {
        stubs: {
          UButton: true,
          UIcon: true
        }
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
