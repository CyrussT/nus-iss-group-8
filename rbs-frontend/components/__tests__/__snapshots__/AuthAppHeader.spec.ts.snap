// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > renders correctly 1`] = `
"<header class="h-16 border-b px-4 flex items-center justify-between">
  <h1 class="font-bold text-xl"><a href="/">Resource Booking System</a></h1><!-- Navigation -->
  <div class="flex items-center gap-4">
    <nav class="relative w-full flex items-center justify-center">
      <ul class="flex items-center min-w-0"></ul>
    </nav>
  </div><!-- Right controls -->
  <div class="flex items-center gap-4"><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-primary-500 dark:text-primary-400 hover:bg-primary-50 disabled:bg-transparent aria-disabled:bg-transparent dark:hover:bg-primary-950 dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center">
      <!--v-if--><span class="iconify i-heroicons:sun w-5 h-5" aria-hidden="true"></span>
      <!--v-if-->
    </button><button type="button" class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-full text-sm gap-x-2.5 px-3.5 py-2.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center">
      <!--v-if--> Logout
      <!--v-if-->
    </button></div>
</header>"
`;

exports[`AuthAppHeader > renders correctly 1`] = `
"<header class="h-16 border-b px-4 flex items-center justify-between">
  <h1 class="font-bold text-xl"><a href="/">Resource Booking System</a></h1><!-- Navigation -->
  <div class="flex items-center gap-4">
    <nav class="relative w-full flex items-center justify-center">
      <ul class="flex items-center min-w-0"></ul>
    </nav>
  </div><!-- Right controls -->
  <div class="flex items-center gap-4"><button type="button" class="focus:outline-none focus-visible:outline-0 disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-md text-sm gap-x-1.5 px-2.5 py-1.5 text-primary-500 dark:text-primary-400 hover:bg-primary-50 disabled:bg-transparent aria-disabled:bg-transparent dark:hover:bg-primary-950 dark:disabled:bg-transparent dark:aria-disabled:bg-transparent focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 inline-flex items-center">
      <!--v-if--><span class="iconify i-heroicons:sun w-5 h-5" aria-hidden="true"></span>
      <!--v-if-->
    </button><button type="button" class="focus:outline-none disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:cursor-not-allowed aria-disabled:opacity-75 flex-shrink-0 font-medium rounded-full text-sm gap-x-2.5 px-3.5 py-2.5 shadow-sm text-white dark:text-gray-900 bg-primary-500 hover:bg-primary-600 disabled:bg-primary-500 aria-disabled:bg-primary-500 dark:bg-primary-400 dark:hover:bg-primary-500 dark:disabled:bg-primary-400 dark:aria-disabled:bg-primary-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-primary-500 dark:focus-visible:outline-primary-400 inline-flex items-center">
      <!--v-if--> Logout
      <!--v-if-->
    </button></div>
</header>"
`;
